lib/rules/func-name-matching.js

```diff
@@ -0,0 +1,148 @@
+/**
+ * @fileoverview Rule to require function names to match the name of the variable or property to which they are assigned.
+ * @author Annie Zhang,
```

Comment:
It seems code is in many ways inspired by the one from JSCS, can give a bit of credit to @xaka too?